tests/services/user.service.spec.ts

```diff
@@ -1,62 +1,35 @@
-import { of } from 'rxjs';
-import { User } from '../../src/app/core/models/user.model';
+import { GithubService } from '../../src/app/core/services/github.service';
+import { LoggingService } from '../../src/app/core/services/logging.service';
 import { UserService } from '../../src/app/core/services/user.service';
-import { jsonData, USER_JUNWEI, USER_Q, USER_SHUMING, USER_WITH_TWO_ROLES } from '../constants/data.constants';
+import { USER_JUNWEI } from '../constants/data.constants';
 
-let dataService: any;
+let githubServiceSpy: jasmine.SpyObj<GithubService>;
+let loggingServiceSpy: jasmine.SpyObj<LoggingService>;
+let userService: UserService;
 
 describe('UserService', () => {
-  describe('.createUserModel(loginId)', () => {
-    beforeAll(() => {
-      dataService = jasmine.createSpyObj('DataService', ['getDataFile']);
-      dataService.getDataFile.and.returnValue(of(jsonData));
-    });
-
-    it('creates a Student user correctly', async () => {
-      await createAndVerifyUser(USER_JUNWEI.loginId, USER_JUNWEI);
-    });
-
-    it('creates a Tutor user correctly when loginId is very short', async () => {
-      await createAndVerifyUser(USER_Q.loginId, USER_Q);
-    });
-
-    it('creates an Admin user correctly', async () => {
-      await createAndVerifyUser(USER_SHUMING.loginId, USER_SHUMING);
-    });
+  beforeAll(() => {
+    githubServiceSpy = jasmine.createSpyObj('GithubService', ['fetchAuthenticatedUser']);
+    loggingServiceSpy = jasmine.createSpyObj('LoggingService', ['info']);
+    userService = new UserService(githubServiceSpy, loggingServiceSpy);
+  });
 
-    it('assigns highest possible role to a user who has multiple roles in data.csv', async () => {
-      await createAndVerifyUser(USER_WITH_TWO_ROLES.loginId, USER_WITH_TWO_ROLES);
+  describe('createUserModel(loginId)', () => {
+    it('should create a user correctly', async () => {
+      const createdUser = await userService.createUserModel(USER_JUNWEI.loginId).toPromise();
+      expect(createdUser).toEqual(USER_JUNWEI);
     });
+  });
 
-    it('should authorize User despite loginId being of different casing', () => {
-      const userService = new UserService(null, dataService);
-      userService.createUserModel(USER_JUNWEI.loginId).subscribe((user) => {
-        expect(user).toBeDefined();
-      });
+  describe('reset()', () => {
+    it('should set current user as undefined', async () => {
+      await userService.createUserModel(USER_JUNWEI.loginId).toPromise();
+      expect(userService.currentUser).toEqual(USER_JUNWEI);
```

Comment:
Resetting a user to undefined feels weird, although as per the specifications given in User Service, aligns with implementation. Might be worth to change this in a separate issue, to something like null. Null has a meaning which aligns much more closely with what we are trying to achieve here.